### PR TITLE
PROTON-1651 Fix typos found by the spellchecker

### PR DESCRIPTION
**Pull request by jdanekrh**
> injecter  proton.Injecter
* injector, maybe. this is public api, so probably should not be touched, though

> Now that we've covered the basics with the archetypical hello world app, let's look at some more interesting examples.

* archetypal, maybe?
* decreffed
** how many f's, proton has f and ff both

> ractor = Reactor.wrap(self.reactor_impl)
* intentional?

I've been probably overzealous what I now realise is just british spelling. Found that deserialised with s is just alternative spelling, I can revert these instances...

### CODE REVIEW

File: examples/cpp/multithreaded_client.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@
 #include <string>
 #include <thread>
 
-// Lock output from threads to avoid scramblin
+// Lock output from threads to avoid scrambling
```
Comment (jdanekrh) [inline examples/cpp/multithreaded_client.cpp]: Is this correct fix?

File: proton-c/bindings/cpp/src/interop_test.cpp [hunk 2]
```diff
@@ -57,8 +57,8 @@ void test_data_ostream() {
     ASSERT_EQUAL("true, false, 42, 42, -42, 12345, -12345, 12345, -12345, 0.125, 0.125", str(dt));
 }
 
-// Test extracting to exact AMQP types works corectly, extrating to invalid types fails.
-void test_decoder_primitves_exact() {
+// Test extracting to exact AMQP types works correctly, extracting to invalid types fails.
+void test_decoder_primitives_exact() {
```
Comment (jdanekrh) [inline proton-c/bindings/cpp/src/interop_test.cpp]: highlighting change to code

File: proton-c/bindings/javascript/module.js [hunk 3]
```diff
@@ -327,7 +327,7 @@ Module.EventDispatch = new function() { // Note the use of new to create a Singl
              * actually free and remove the selectable is deferred until next
              * time round the (internal JavaScript) event loop. This turned out
              * to be necessary because in some cases the ws WebSocket library
-             * calls the onclose callback (concurrently!!) before the onmessage
+             * calls the on_close callback (concurrently!!) before the on_message
```
Comment (jdanekrh) [inline proton-c/bindings/javascript/module.js]: couldn't find what are correct names for these callbacks, maybe I got it wrong

File: proton-c/bindings/ruby/lib/core/connection.rb [hunk 4]
```diff
@@ -201,7 +201,7 @@ def remote_desired_capabilities
     # @return [Data] The remote properties.
     #
     def remote_properties
-      data_to_object(Cproton.pn_connection_remote_properites(@impl))
+      data_to_object(Cproton.pn_connection_remote_properties(@impl))
```
Comment (jdanekrh) [inline proton-c/bindings/ruby/lib/core/connection.rb]: highlighting code change

File: proton-c/bindings/ruby/lib/core/ssl.rb [hunk 5]
```diff
@@ -126,25 +126,25 @@ def cipher_name
...
-    # time due to renegotation.
+    # returns nil if SSL is not active. Not that the protocol may change over
+    # time due to renegotiation.
     #
     # @return [String, nil] The protocol name.
     #
     def protocol_name
       rc, name = Cproton.pn_ssl_get_protocol_name(@impl, 128)
-      retur name if rc
+      return name if rc
```
Comment (jdanekrh) [inline proton-c/bindings/ruby/lib/core/ssl.rb]: highlighting code change

File: proton-c/bindings/ruby/lib/handler/endpoint_state_handler.rb [hunk 6]
```diff
@@ -164,7 +164,7 @@ def on_link_error(event)
         Qpid::Proton::Event.dispatch(@delegate, :on_link_error, event)
       else
         self.log_error(event.link, "link")
-        event.conneciton.close
+        event.connection.close
```
Comment (jdanekrh) [inline proton-c/bindings/ruby/lib/handler/endpoint_state_handler.rb]: higlighting code change

File: proton-c/bindings/ruby/lib/reactor/container.rb [hunk 7]
```diff
@@ -155,7 +155,7 @@ def create_sender(context, opts = {})
         sender.source.address = opts[:source] if !opts[:source].nil?
         sender.target.address = target if target
         sender.handler = opts[:handler] if !opts[:handler].nil?
-        sender.tag_generator = opts[:tag_generator] if !opts[:tag_gnenerator].nil?
+        sender.tag_generator = opts[:tag_generator] if !opts[:tag_generator].nil?
```
Comment (jdanekrh) [inline proton-c/bindings/ruby/lib/reactor/container.rb]: highlighting code change

File: tests/python/proton_tests/message.py [hunk 8]
```diff
@@ -69,7 +69,7 @@ def testPriority(self):
   def testTtl(self):
     self._test("ttl", 0, range(12345, 54321))
 
-  def testFirstAquirer(self):
+  def testFirstAcquirer(self):
```
Comment (jdanekrh) [inline tests/python/proton_tests/message.py]: highlighting code change

File: tests/python/proton_tests/messenger.py [hunk 9]
```diff
@@ -629,7 +629,7 @@ def testDefaultRewriteSUHPN(self):
   def testDefaultRewriteSUPHPN(self):
     self._testRewrite("amqp://user:pass@original:123/name", "amqp://original:123/name")
 
-  def testRewriteSupress(self):
+  def testRewriteSuppress(self):
```
Comment (jdanekrh) [inline tests/python/proton_tests/messenger.py]: highlighting code chabge

**Comment by jdanekrh** [comment]
What is blech?

 >   transport_consume(transport);        // blech - testBindAfterOpen

File: proton-c/bindings/cpp/src/messaging_adapter.cpp [hunk 10]
```diff
@@ -275,7 +275,7 @@ void on_transport_closed(messaging_handler& handler, pn_event_t* event) {
     // If the connection isn't open generate on_transport_open event
     // because we didn't generate it yet and the events won't match.
     pn_connection_t *conn = pn_event_connection(event);
-    if (!conn || is_remote_unititialised(pn_connection_state(conn))) {
+    if (!conn || is_remote_uninitialized(pn_connection_state(conn))) {
```
Comment (jdanekrh) [inline proton-c/bindings/cpp/src/messaging_adapter.cpp]: highlighting code change

**Comment by ssorj** [comment]
@jdanekrh , I adapted this patch and applied it.  I excluded any diffs against the Ruby binding since it's been revamped.  If you're satisified, please close.

https://github.com/apache/qpid-proton/commit/e22ca536c69e840bbe9816029a5903408c9a2570

**Comment by jdanekrh** [comment]
I am ok leaving injecter be. I still have these three suggested changes open (which I only mentioned in the discussion but weren't part of the PR). I'll mention them on a new ticket, If I ever do another spellcheck pass over the codebase.

> Now that we've covered the basics with the archetypical hello world app, let's look at some more interesting examples.

archetypal, maybe?

> decreffed

how many f's? this will require some English language skills, because proton has instances of f and ff both and spellcheckers recognize neither variant

> transport_consume(transport);        // blech - testBindAfterOpen

What is blech?

**Pull request closed without merging**